File: backend/models/Event.js
const mongoose = require('mongoose');

const eventSchema = new mongoose.Schema({
  title: {
    type: String,
    required: true,
    trim: true
  },
  description: {
    type: String,
    required: true
  },
  flyerImageUrl: {
    type: String,
    default: ''
  },
  date: {
    type: Date,
    required: true
  },
  venue: {
    type: String,
    required: true
  },
  pricing: {
    visitorPrice: {
      type: Number,
      default: 0
    },
    memberPrice: {
      type: Number,
      default: 0
    }
  },
  participants: [{
    user: {
      type: mongoose.Schema.Types.ObjectId,
      ref: 'User'
    },
    registeredAt: {
      type: Date,
      default: Date.now
    }
  }],
  createdBy: {
    type: mongoose.Schema.Types.ObjectId,
    ref: 'User',
    required: true
  },
  createdAt: {
    type: Date,
    default: Date.now
  }
}, {
  timestamps: true
});

module.exports = mongoose.model('Event', eventSchema);

File: ECGCommunityApp/ECGCommunityApp/Models/Event.swift
import Foundation

struct Event: Codable, Identifiable {
    let id: String
    let title: String
    let description: String
    let flyerImageUrl: String?
    let date: Date
    let venue: String
    let pricing: EventPricing
    let participants: [EventParticipant]
    let createdBy: EventCreator
    let isParticipating: Bool?
    
    enum CodingKeys: String, CodingKey {
        case id = "_id"
        case title, description, flyerImageUrl, date, venue, pricing, participants, createdBy, isParticipating
    }
}

struct EventPricing: Codable {
    let visitorPrice: Double
    let memberPrice: Double
}

struct EventParticipant: Codable, Identifiable {
    let id: String
    let user: User
    let registeredAt: Date
    
    enum CodingKeys: String, CodingKey {
        case id = "_id"
        case user, registeredAt
    }
}

struct EventCreator: Codable, Identifiable {
    let id: String
    let username: String
    
    enum CodingKeys: String, CodingKey {
        case id = "_id"
        case username
    }
}

